lib/extractor/index.ts

```diff
@@ -38,15 +38,43 @@ async function getDockerArchiveLayersAndManifest(
   };
 }
 
-function getContent(
+function isBufferType(type: string | Buffer): type is Buffer {
+  return (type as Buffer).buffer !== undefined;
+}
+
+function isStringType(type: string | Buffer): type is string {
+  return (type as string).substring !== undefined;
+}
+
+function getContentAsBuffer(
+  extractedLayers: ExtractedLayers,
+  extractAction: ExtractAction,
+): Buffer | undefined {
+  const content = getContent(extractedLayers, extractAction);
+  return content !== undefined && isBufferType(content) ? content : undefined;
```

Comment:
why do we return undefined instead of throwing an exception? I don't like returning undefined in case of errors. it looks like C programming :)